src/tls/25519.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "pem.h"
+#include "tls.h"
+
+namespace tls
+{
```

Comment:
are these tested? also would it provide additional information on the errors thrown to add the failure code?

Reply:
Good point! I added a test for it and include the `tls::error_string(rc)` throughout the file.